code/game/objects/items/food/burgers.dm

```diff
@@ -39,12 +39,13 @@
 /obj/item/food/burger/human/CheckParts(list/parts_list)
 	..()
 	var/obj/item/food/patty/human/human_patty = locate(/obj/item/food/patty/human) in contents
-	if(LAZYLEN(human_patty.custom_materials))
-		for(var/datum/material/meat/mob_meat/mob_meat_material in human_patty.custom_materials)
-			if(mob_meat_material.subjectname)
-				name = "[mob_meat_material.subjectname] burger"
-			else if(mob_meat_material.subjectjob)
-				name = "[mob_meat_material.subjectjob] burger"
+	if(!LAZYLEN(human_patty.custom_materials))
```

Comment:
for(var/element in some_list) doesnt iterate if the list is empty or null so theres no need for this length check